pkg/armhelpers/azureclient.go

```diff
@@ -0,0 +1,313 @@
+package armhelpers
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/acs-engine/pkg/acsengine"
+
+	"github.com/Azure/azure-sdk-for-go/arm/authorization"
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/Azure/azure-sdk-for-go/arm/resources/resources"
+	"github.com/Azure/azure-sdk-for-go/arm/resources/subscriptions"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/to"
+	log "github.com/Sirupsen/logrus"
+	"github.com/mitchellh/go-homedir"
+)
+
+const (
+	// AcsEngineClientID is the AAD ClientID for the CLI native application
+	AcsEngineClientID = "76e0feec-6b7f-41f0-81a7-b1b944520261"
+
+	// ApplicationDir is the name of the dir where the token is cached
+	ApplicationDir = ".acsengine"
+)
+
+var (
+	// RequiredResourceProviders is the list of Azure Resource Providers needed for ACS-Engine to function
+	RequiredResourceProviders = []string{"Microsoft.Compute", "Microsoft.Storage", "Microsoft.Network"}
+)
+
+// AzureClient is the uber client
+// If done right, we really shouldn't need SubscriptionID or TenantID in the client anywhere else
+// they're only used for setting up the clients, we can add them back later if needed
+type AzureClient struct {
```

Comment:
The more I think about this, the more I don't want to do it. It's not common for struct types to be pluralized and seems a bit misleading. What would it mean to have an array of `AzureClients`, etc?